src/pages/allPosts/AllPosts.jsx

```diff
@@ -1,13 +1,75 @@
 import './AllPosts.css';
+import {Link, useParams} from "react-router-dom";
+
+import axios from "axios";
+import {useEffect} from "react";
+import {useState} from "react";
 
 function AllPosts() {
 
-    return(
+    const [userPosts, setPosts] = useState([])
+    const [error, setError] = useState(null)
+
+    useEffect(() => {
+
+        async function getPosts() {
```

Comment:
als je de json importeerd dan kan je via Link-to direct navigeren naar de juiste post en wordt de code een stuk simpeler